src/main/java/org/openlmis/fulfillment/web/OrderController.java

```diff
@@ -192,15 +190,14 @@ public OrderDto createOrder(@RequestBody OrderDto orderDto,
   @ResponseStatus(HttpStatus.OK)
   @ResponseBody
   public Page<BasicOrderDto> searchOrders(OrderSearchParams params, Pageable pageable) {
-    List<Order> orders = orderService
-        .searchOrders(params)
-        .stream()
-        .filter(Objects::nonNull)
-        .sorted((o1, o2) -> ObjectUtils.compare(o1.getId(), o2.getId()))
-        .collect(Collectors.toList());
+    List<Order> orders = orderService.searchOrders(params);
 
     List<Order> filteredList = orderSecurityService.filterInaccessibleOrders(orders);
-    Iterable<BasicOrderDto> data = BasicOrderDto.newInstance(filteredList, exporter);
+    // with this we will only create basic order dto objects for elements that would be
+    // returned by this endpoint
+    Page<Order> page = Pagination.getPage(filteredList, pageable);
```

Comment:
The biggest performance gain would be doing the pagination on the db, not here statically. Or do we have the same permissions issue that happens in requisitions?